tools/python_tools/cuvslam_tools/dataset_registry.py

```
@@ -0,0 +1,372 @@
+    CUVSLAM_DATASETS. If that disagrees with the dataset ID, the reporter looks
+    for sequences in a directory staging never created.
+    """
+    expected = f"{spec.mount_name}/"
+    for record in spec.evals:
+        config_path = root / record.config
+        if not config_path.is_file():
+            raise RegistryError(f"{spec.dataset_id}: staged config not found: {config_path}")
+        try:
+            declared = json.loads(config_path.read_text(encoding="utf-8")).get("dataset_folder")
+        except json.JSONDecodeError as exc:
+            raise RegistryError(f"{spec.dataset_id}: {config_path} is not valid JSON: {exc}") from exc
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)